src/host/proxy/ITerminalHandoff.idl

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import "oaidl.idl";
+import "ocidl.idl";
+
+[
+    object,
+    uuid(FA1E3AB4-9AEC-4A3C-96CA-E6078C30BD74)
+] interface ITerminalHandoff : IUnknown
+{
+    HRESULT EstablishHandoff([in, system_handle(sh_pipe)] HANDLE in,
+                             [in, system_handle(sh_pipe)] HANDLE out,
+                             [in, system_handle(sh_pipe)] HANDLE signal);

```

Comment:
Interface concern: I think we should pass the process name (if we can!) _and the process handle_.

* **why the handle?** ConptyConnection wants to wait on its process handle so that it can get the exit code. It would not feel right for def-term-launched processes to not get the same "process exited with code XXX" treatment.
* **why the name?** eventually we will want to let people customize which profiles get assigned to which incoming apps.

Reply:
I added handle, but I'm not sure which name you mean. Can we not just query the name from the handle instead of marshalling a string?

Reply:
We will address the rest of this in #9458